alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/accessor/DefaultDescriptorGlobalConfigUtility.java

```diff
@@ -96,9 +98,33 @@ public FieldModel save(FieldModel fieldModel) throws AlertException {
 
     public FieldModel update(Long id, FieldModel fieldModel) throws AlertException {
         FieldModel beforeUpdateAction = apiAction.beforeUpdateAction(fieldModel);
-        Collection<ConfigurationFieldModel> values = configurationFieldModelConverter.convertToConfigurationFieldModelMap(beforeUpdateAction).values();
-        ConfigurationModel configurationModel = configurationAccessor.updateConfiguration(id, values);
+        Map<String, ConfigurationFieldModel> valueMap = configurationFieldModelConverter.convertToConfigurationFieldModelMap(beforeUpdateAction);
+        Optional<ConfigurationModel> existingConfig = configurationAccessor.getConfigurationById(id);
+        ConfigurationModel configurationModel;
+        if (existingConfig.isPresent()) {
+            Map<String, ConfigurationFieldModel> updatedValues = updateSensitiveFields(valueMap, existingConfig.get());
+            configurationModel = configurationAccessor.updateConfiguration(id, updatedValues.values());
+        } else {
+            configurationModel = configurationAccessor.createConfiguration(key, context, valueMap.values());
+        }
         FieldModel convertedFieldModel = configurationFieldModelConverter.convertToFieldModel(configurationModel);
         return apiAction.afterUpdateAction(convertedFieldModel);
     }
+
+    private Map<String, ConfigurationFieldModel> updateSensitiveFields(Map<String, ConfigurationFieldModel> values, ConfigurationModel existingConfig) {
```

Comment:
I'm surprised we don't have a utility for this. We should probably add a TODO here and an issue for 6.1.0

Reply:
FieldModelProcessor will handle these changes.  The problem is that FieldModelProcessor is in the alert-main project and this utility is in the alert-common sub project.  Using FieldModelProcessor would create a dependency we don't want.  If we can move FieldModelProcessor into alert-common then that would resolve this issue.  I will add a TODO comment.